python/dgl/graphbolt/impl/ondisk_dataset.py

```diff
@@ -16,13 +32,186 @@
+            # Always save the feature in numpy format.
+            out_feature["format"] = "numpy"
+            out_feature["path"] = processed_dir_prefix / feature[
+                "path"
+            ].replace("pt", "npy")
+
+            if feature["format"] == "torch":
+                data = read_data(
+                    dataset_path / feature["path"],
+                    feature["format"],
+                    in_memory=feature["in_memory"],
+                )
+                save_data(
+                    data,
+                    dataset_path / out_feature["path"],
+                    out_feature["format"],
+                )
+            elif feature["format"] == "numpy":
+                # If the original format is numpy, just copy the file.
+                os.makedirs(
+                    dataset_path / os.path.dirname(out_feature["path"]),
+                    exist_ok=True,
+                )
+                shutil.copyfile(
+                    dataset_path / feature["path"],
+                    dataset_path / out_feature["path"],
+                )
+            else:
+                raise ValueError(
+                    f"Unsupported feature format {feature['format']}"
+                )
+
+    # 7. Save the train/val/test split according to the output_config.
+    for set_name in ["train_sets", "validation_sets", "test_sets"]:
+        if set_name not in input_config:
+            continue
+        for intput_set_split, output_set_split in zip(
+            input_config[set_name], output_config[set_name]
+        ):
+            for input_set_per_type, output_set_per_type in zip(
+                intput_set_split, output_set_split
+            ):
+                # Always save the feature in numpy format.
+                output_set_per_type["format"] = "numpy"
+                output_set_per_type[
+                    "path"
+                ] = processed_dir_prefix / input_set_per_type["path"].replace(
+                    "pt", "npy"
+                )
+                if input_set_per_type["format"] != "numpy":
+                    # If the original format is not numpy, convert it to numpy.
+                    input_set = read_data(
+                        dataset_path / input_set_per_type["path"],
+                        input_set_per_type["format"],
+                    )
+                    save_data(
+                        input_set,
+                        dataset_path / output_set_per_type["path"],
+                        output_set_per_type["format"],
+                    )
+                else:
+                    os.makedirs(
+                        dataset_path
+                        / os.path.dirname(output_set_per_type["path"]),
+                        exist_ok=True,
+                    )
+                    shutil.copy(
+                        dataset_path / input_set_per_type["path"],
+                        dataset_path / output_set_per_type["path"],
+                    )
```

Comment:
```suggestion
                if input_set_per_type["format"] == "torch":
                    # If the original format is not numpy, convert it to numpy.
                    input_set = read_data(
                        dataset_path / input_set_per_type["path"],
                        input_set_per_type["format"],
                    )
                    save_data(
                        input_set,
                        dataset_path / output_set_per_type["path"],
                        output_set_per_type["format"],
                    )
                elif input_set_per_type["format"] == "numpy":
                    os.makedirs(
                        dataset_path
                        / os.path.dirname(output_set_per_type["path"]),
                        exist_ok=True,
                    )
                    shutil.copy(
                        dataset_path / input_set_per_type["path"],
                        dataset_path / output_set_per_type["path"],
                    )
                else:
                    raise ValueError(
                        f"Unsupported feature format {feature['format']}"
                    )
```

Reply:
I think we can make full use of the internal code of `read_ data`:
```
def read_data(path, fmt, in_memory=True):
    """Read data from disk."""
    if fmt == "torch":
        return _read_torch_data(path)
    elif fmt == "numpy":
        return _read_numpy_data(path, in_memory=in_memory)
    else:
        raise RuntimeError(f"Unsupported format: {fmt}")
```

I will modify the logic and only make a special judgment if `copy` is needed.